servers/rendering/renderer_rd/storage_rd/mesh_storage.cpp

```diff
@@ -1307,22 +1304,32 @@ bool MeshStorage::_multimesh_enable_motion_vectors(RID p_multimesh) {
 		multimesh->data_cache.append_array(multimesh->data_cache);
 	}
 
-	if (multimesh->buffer_set) {
+	uint32_t buffer_size = multimesh->instances * multimesh->stride_cache * sizeof(float);
+	uint32_t new_buffer_size = buffer_size * 2;
+	RID new_buffer = RD::get_singleton()->storage_buffer_create(new_buffer_size);
+
+	if (multimesh->buffer_set && multimesh->data_cache.is_empty()) {
+		// If the buffer was set but there's no data cached in the CPU, we must download it from the GPU and
+		// upload it because RD does not provide a way to copy the buffer directly yet.
 		RD::get_singleton()->barrier();
 		Vector<uint8_t> buffer_data = RD::get_singleton()->buffer_get_data(multimesh->buffer);
-		if (!multimesh->data_cache.is_empty()) {
-			memcpy(buffer_data.ptrw(), multimesh->data_cache.ptr(), buffer_data.size());
-		}
+		ERR_FAIL_COND(buffer_data.size() != int(buffer_size));
+		RD::get_singleton()->buffer_update(new_buffer, 0, buffer_size, buffer_data.ptr(), RD::BARRIER_MASK_NO_BARRIER);
+		RD::get_singleton()->buffer_update(new_buffer, buffer_size, buffer_size, buffer_data.ptr());
+
+
+	} else if (!multimesh->data_cache.is_empty()) {
+		// Simply upload the data cached in the CPU, which should already be doubled in size.
+		ERR_FAIL_COND(multimesh->data_cache.size() != int(new_buffer_size));
+		RD::get_singleton()->buffer_update(new_buffer, 0, new_buffer_size, multimesh->data_cache.ptr());
+	}
 
+	if (multimesh->buffer.is_valid()) {
 		RD::get_singleton()->free(multimesh->buffer);
-		uint32_t buffer_size = multimesh->instances * multimesh->stride_cache * sizeof(float) * 2;
-		multimesh->buffer = RD::get_singleton()->storage_buffer_create(buffer_size);
-		RD::get_singleton()->buffer_update(multimesh->buffer, 0, buffer_data.size(), buffer_data.ptr(), RD::BARRIER_MASK_NO_BARRIER);
-		RD::get_singleton()->buffer_update(multimesh->buffer, buffer_data.size(), buffer_data.size(), buffer_data.ptr());
-		multimesh->uniform_set_3d = RID(); // Cleared by dependency
-		return true;
 	}
-	return false; // Update the transforms uniform set cache
+	
+	multimesh->buffer = new_buffer;
+	multimesh->uniform_set_3d = RID(); // Cleared by dependency
```

Comment:
```suggestion
	multimesh->uniform_set_3d = RID(); // Cleared by dependency.
```

Reply:
Should clarify that's copy pasted from the previous version and the comment is repeated elsewhere in the file. Should we fix it in this particular PR or another one for the entire file itself?

Reply:
Just fix this exact line IMO, when a comment is part of the code you've added/changed/copied it should be correct, leave the rest as is

Reply:
For clarification. We ask contributors to update the style of comments that they touch because we don't want to do one giant PR to fix the style of comments. Doing so would really hurt the readability of our git history.